### DISCUSSION TITLE
Empty Configuration List

### DISCUSSION THREAD

**Discussion by JohnLeyenaar** [arkypita/LaserGRBL#2156] [General]
With machine connected and Idle, I request Gbrl Configuration but the list is empty. Shows "0  Config read successfully"
If I enter "$$" in the command line I do see a list of parameters in the sidebar.
Is this a problem with the software, a problem with interface to the machine (Latitool L50) or do I need to do something?

**Comment by arkypita** [comment]
It is probably a mix. Machine send data with something that make it unrecognizable for LaserGRBL parsing logic.
I'll thank you if you can do some debugging for me so I'll be able to fix it.

Can you perform a session with extended log activated and upload your comlog.txt here?

Open LaserGRBL
Menu "?" activate extended log
Chose where to save
Connect the engraver
Open GRBL configuration form
Wait until "zero config read is shown"
Press button "read" to force a second read
Close the form
Disconnect

Then upload comlog.txt here and please also upload "session log" (Menu "?" - "open session log")

 Be sure to use the latest LaserGRBL version 5.6.0 https://github.com/arkypita/LaserGRBL/releases/download/v5.6.0/install.exe

**Reply by JohnLeyenaar** [reply]
Okay Thanks. Here are the files:
[comlog.txt](https://github.com/arkypita/LaserGRBL/files/13394605/comlog.txt)
[sessionlog.txt](https://github.com/arkypita/LaserGRBL/files/13394606/sessionlog.txt)

I just got the machine and LaserGBRL this week (Nice job on the software by the way!). Quite easy to get everything running.
Not sure what version I have (normally I would look for an "about" selection to find out).
I did just check for updates though, and it said I have the latest.

**Reply by arkypita** [reply]
You have 5.4.0, you can see it in splash screen and read in session log.
Could you please download 5.6.0 that has improved diagnostic useful to understand this issue?

https://github.com/arkypita/LaserGRBL/releases/download/v5.6.0/install.exe

Than re-do the test

**Reply by arkypita** [reply]
Edit: You have 5.4.0

**Reply by JohnLeyenaar** [reply]
Okay, I installed Version 5.6.0 (It shows as 5.6.0 on the splash screen and in upper left corner next to LaserGRBL).
Ran the test again and here are the files (I noticed that the session log still shows 5.4.0 ?):

[comlog.txt](https://github.com/arkypita/LaserGRBL/files/13398603/comlog.txt)
[sessionlog.txt](https://github.com/arkypita/LaserGRBL/files/13398604/sessionlog.txt)

**Reply by arkypita** [reply]
Dis you still have empty configuration and 0 config read?

**Reply by JohnLeyenaar** [reply]
Yes, still empty configuration list. I will try it again later today.

**Reply by JohnLeyenaar** [reply]
Tried again and still empty configuration list

**Reply by arkypita** [reply]
Ok, thank you for your feedback.
That's strange because session log say  

17-11-2023 21:07:01.679	Refresh Config	Refreshing grbl configuration [OnConnect]
17-11-2023 21:07:01.748	Refresh Config	Configuration successfully received! (Expected: 50 Found: 23)

(Note 50 and 23 are inverted in this log message)

So you should see all 50 configuration flags that your engraver has.

Now I am on vacation, I'll do some testing when I am back and let you know.

Ps: you should read the version 5.6.0 in session log in last lines, because this is an history log.

**Reply by JohnLeyenaar** [reply]
Hope you are having a good vacation!

Just a thought for when you get back:
It is as if the parameters are being read, but not being saved (cached). I say this because another thing I noticed is that when I create a job I will get a warning that the size is too large for the values in $130 and $131. It thinks $130=300 and $131=200 (see pic).
But actually (as you can see in the comlog), $130=380 and $131=380. I had changed these values earlier using the command line. I simply ignore the warning and continue and all is good.
Perhaps the list is reading from where the values should have been saved (cached), but nothing is there, in the same way it doesn't see the true values for $130 and $131?

![Screenshot 2023-11-20 064538](https://github.com/arkypita/LaserGRBL/assets/12695559/25b1b04d-2341-4785-9ffe-3d235578061f)

**Comment by arkypita** [comment]
Hi @JohnLeyenaar 

Thanks for sharing your suggestions.  $130=300 and $131=200 are default values used by low level read for width and height that LaserGRBL use in case the configuration from the machine are not available/readable or convertibile to a number (ie. if configuration is damaged or wrong).

I've tried to simulate your situation extensively, and I've never been able to reproduce it. I really can't understand what the problem could be with your machine, but it seems like a common problem for those who have a "SimpleLaser 1.1h" controller (read [here](https://github.com/bdring/FluidNC/issues/857#issuecomment-1669543638), [here](https://www.facebook.com/groups/lasergrbl/posts/1613246255836031/?__cft__[0]=AZWBuOmujljUGuSaGExAby8qz8JluGJbCBCw-_GNGR30aeODpu2dYwZP4v9seaW2B50xaSFWZLbxjSBXL_78BT9bjT9L4LtPili_M689Kfi_rpDgRxfXcdoOBjOn_s6Yw_tyxZoGZ27dVxy77nTJ6Q3-bEYkLecO4jsM3v5J_4NlbNMUZacMe0yVoWtkyTST2DU&__tn__=%2CO%2CP-R) and [here](https://www.facebook.com/groups/lasergrbl/posts/1646997429127580/?__cft__[0]=AZUYvXRkeEHCh43InHQFEXTXsSyfgwgRBEE7v5kIrQUop7MSO8f6rO68J8XPTHSKxbsDmKSe9tm8KeEVqAy9Ey1ImaZQMPglHSsWxSff4fDZ0vlQJ4qdLUPmwT1_6VW6OanNo37_eZR5lU2nil90npSU4WiUhWRbhs7fAKGJ2cSSAqsK6_8Zbv9S1VXuLkBRoLI&__tn__=%2CO%2CP-R) and [here](https://www.facebook.com/photo/?fbid=10232203613219288&set=gm.1630263144134342&idorvanity=486886768471991) too)

"Simple Laser v1.1h" (that is the firmware your machine controller is using) is not an original Grbl firmware, but a 3rd party GRBL clone probably not 100% compatible with the GRBL specification. I read that some have solved the same problem as yours with a firmware update, but I would like to be able to understand what the problem is and solve it on LaserGRBL's side so that I can also manage machines with this version of firmware (and if you help me on this, I thank you).

Everything in the communication log makes it seem like it should be working, but there should be something I'm missing / not understanding. I made a version of LaserGRBL that has even more diagnostic messages, so I can delve deeper into what's going on.

It is 5.6.1 and you can download it from here: [install.zip](https://github.com/arkypita/LaserGRBL/files/13463949/install.zip)


![image](https://github.com/arkypita/LaserGRBL/assets/8782035/3fb38b2f-b7c0-4a3b-bc41-8c79978ebd85)

**Reply by JohnLeyenaar** [reply]
Thanks @arkypita

Well I had a more serious problem with the Engraver - After only about four hours of actual engraving, the Laser module failed.
I tried a number of things but no luck. I am in the process of returning it for a refund. So unfortunately I won't be able to help much with this particular problem anymore. Sorry things worked out this way because I realize you are interested in finding the problem.
I do think you are correct about a problem with the version. I saw somewhere online (but I can't find it again) that version "Simple Laser v1.1k" seems to correct the problem. Don't know for sure of coarse (I can't even find out where I could have gotten that version).

I am getting a different Engraver. It should be here in a few days. Getting a Creality CR-Laser 10W which is a more common brand (I think they are big in 3D Printers). Also much better online support. I will let you know how this new one works out with LaserGBRL.
I will install the newer version you sent anyway - Thanks

Thanks again for all your help!!

**Reply by JohnLeyenaar** [reply]
@arkypita ,

Got my new machine and it works much better with LaserGRBL. No problem with config list.
This is what it looks like on startup:
![Screenshot 2023-11-28 063432](https://github.com/arkypita/LaserGRBL/assets/12695559/062c0aaa-884f-4375-9422-6d92046adc96)

**Comment by arkypita** [comment]
Fixed in version v5.7
